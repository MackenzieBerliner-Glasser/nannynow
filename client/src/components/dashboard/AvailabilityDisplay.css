#availabilityList {
  margin: 15px;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}