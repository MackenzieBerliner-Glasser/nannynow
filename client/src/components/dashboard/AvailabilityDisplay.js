import React, { Fragment, PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './AvailabilityDisplay.css';
import Loading from '../loading/Loading';

export default class AvailabilityDisplay extends PureComponent {
  static propTypes = {
    availability: PropTypes.array,
    removeAvailability: PropTypes.func.isRequired,
    getAvailability: PropTypes.func.isRequired
  };

  componentDidMount() {
    const { _id } = this.props.user;
    console.log(this.props);
    // add in error guard clause
    if(!this.props.availability) return this.props.getAvailability(_id);
  }

  handleRemove = id => {
    return () => this.props.removeAvailability(id);
  };

  render() {

    const { availability } = this.props;
    if(!availability) return <Loading />;
    const availabilityList = availability.map((available, i) => {
      const { _id, availableStartTime, availableEndTime } = available;
      return (
        <div key={i} id={styles.availabilityList}>
          <div><b>Start Time/Date:</b> {availableStartTime}</div>
          <div><b>End Time Date:</b> {availableEndTime}</div>
          <button onClick={this.handleRemove(_id)}>Remove</button>
        </div>
      );
    });
    return (
      <div>
        <div>Available From:</div>
        <div>Available Until:</div>
        {availabilityList}
      </div>
    );
  }
}
