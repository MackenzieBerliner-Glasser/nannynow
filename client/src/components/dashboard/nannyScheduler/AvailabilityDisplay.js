import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './AvailabilityDisplay.css';
import Loading from '../../loading/Loading';
import { getLocalDateTime } from '../../helpers/time';

export default class AvailabilityDisplay extends PureComponent {
  static propTypes = {
    availability: PropTypes.array,
    removeAvailability: PropTypes.func.isRequired,
    getAvailability: PropTypes.func.isRequired
  };

  componentDidMount() {
    const { _id } = this.props.user;
    if(!this.props.availability) return this.props.getAvailability(_id);
  }

  handleRemove = id => {
    const { _id } = this.props.user;
    return () => this.props.removeAvailability(id)
      .then(this.props.getAvailability(_id));
  };

  render() {

    const { availability } = this.props;
    if(!availability) return <Loading />;
    const availabilityList = availability.map((available, i) => {
      const { _id, availableStartTime, availableEndTime } = available;
      return (
        <div key={i} id={styles.availabilityList}>
          <div><b>Start Time/Date:</b> {getLocalDateTime(availableStartTime)}</div>
          <div><b>End Time Date:</b> {getLocalDateTime(availableEndTime)}</div>
          <button id={styles.removeButton} onClick={this.handleRemove(_id)}>Remove</button>
        </div>
      );
    });
    return (
      <div>
        <h3>Your Upcoming Availability</h3>
        {availabilityList}
      </div>
    );
  }
}
