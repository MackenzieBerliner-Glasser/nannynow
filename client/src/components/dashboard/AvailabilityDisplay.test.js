import React from 'react';
import { shallow } from 'enzyme';
import AvailabilityDisplay from './AvailabilityDisplay';

describe('AvailabilityDisplay', () => {
  let startTime1 = new Date('2018-12-15T16:00:00.000Z');
  let endTime1 = new Date('2018-12-15T21:00:00.000Z');
  let startTime2 = new Date('2018-12-16T20:00:00.000Z');
  let endTime2 = new Date('2018-12-17T02:00:00.000Z');

  const availability = [
    {
      availableStartTime: startTime1,
      availableEndTime: endTime1
    },
    {
      availableStartTime: startTime2,
      availableEndTime: endTime2
    }
  ];

  it('matches snapshot', () => {
    const wrapper = shallow(
      <AvailabilityDisplay availability={availability} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
