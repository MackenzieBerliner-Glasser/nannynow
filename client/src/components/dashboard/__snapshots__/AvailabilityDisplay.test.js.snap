// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvailabilityDisplay matches snapshot 1`] = `
<div>
  <div>
    Available From:
  </div>
  <div>
    Available Until:
  </div>
  <div
    id="availabilityList"
    key="0"
  >
    <div>
      <b>
        Start Time/Date:
      </b>
       
      <Component />
    </div>
    <div>
      <b>
        End Time Date:
      </b>
       
      <Component />
    </div>
    <button
      onClick={[Function]}
    >
      Remove
    </button>
  </div>
  <div
    id="availabilityList"
    key="1"
  >
    <div>
      <b>
        Start Time/Date:
      </b>
       
      <Component />
    </div>
    <div>
      <b>
        End Time Date:
      </b>
       
      <Component />
    </div>
    <button
      onClick={[Function]}
    >
      Remove
    </button>
  </div>
</div>
`;
